src/Agent.Worker/Handlers/HandlerFactory.cs

```diff
@@ -0,0 +1,59 @@
+using Microsoft.VisualStudio.Services.Agent.Util;
+using Microsoft.VisualStudio.Services.Agent.Worker;
+using System;
+using System.Collections.Generic;
+
+namespace Microsoft.VisualStudio.Services.Agent.Worker.Handlers
+{
+    [ServiceLocator(Default = typeof(HandlerFactory))]
+    public interface IHandlerFactory : IAgentService
+    {
+        IHandler Create(
+            IExecutionContext executionContext,
+            HandlerData data,
+            Dictionary<string, string> inputs,
+            string taskDirectory);
+    }
+
+    public sealed class HandlerFactory : AgentService, IHandlerFactory
```

Comment:
The main purpose of this class is: Give it a `HandlerData` object (handler execution section from the task.json), and it creates a new instance for the corresponding handler.

Unfortunately the distributed task SDK doesn't provide the classes we need to simply deserialize the task.json into a big data structure. So I made small data classes for deserializing the handler execution section. I originally started down the path of working with the J objects. After a few iterations, the strongly typed data classes seemed easier to follow. I put the classes in the TaskDefinitionManager.cs file... It's further down in the PR.